packages/babel-parser/src/plugins/typescript/index.js

```diff
@@ -1528,6 +1528,14 @@ export default (superClass: Class<Parser>): Class<Parser> =>
       if (!this.isRelational("<")) {
         return undefined;
       }
+
+      const oldMaybeInArrowParameters = this.state.maybeInArrowParameters;
```

Comment:
I wasn't 100% sure where to put this change, and the one below to reset things. 

For resetting these values, it seemed best to do it before any attempt to parse is started, but strictly speaking it only seemed to need to happen before `parseFunctionParams`.

For setting state back to oldState below, it looks like typically this is done before `parseArrowExpression` is called (e.g. https://github.com/liamfd/babel/blob/master/packages/babel-parser/src/parser/expression.js#L972), so I tried to match that.